kotori/daq/storage/cratedb.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+# (c) 2023 Andreas Motl <andreas@getkotori.org>
+import calendar
+import functools
+import json
+from collections import OrderedDict
+from decimal import Decimal
+from copy import deepcopy
+from datetime import datetime, date, timezone
+
+import crate.client.http
+import pytz
+import requests
+from crate import client
+from crate.client.converter import DefaultTypeConverter
+from crate.client.exceptions import ProgrammingError
+from funcy import project
+from munch import Munch
+from twisted.logger import Logger
+
+from kotori.daq.storage.util import format_chunk
+
+log = Logger()
+
+
+class CrateDBAdapter:
+    """
+    Kotori database backend adapter for CrateDB.
+
+    CrateDB is a distributed SQL database for storing and analyzing
+    massive amounts of data in real-time. Built on top of Lucene.
```

Comment:
Update description/slogan, it changed in the meanwhile.

```suggestion
   CrateDB is a distributed and scalable SQL database for storing and analyzing massive
   amounts of data in near real-time, even with complex queries. It is PostgreSQL-compatible,
   and based on Lucene.
```